src/back/glsl/mod.rs

```diff
@@ -317,6 +317,13 @@ pub struct Writer<'a, W> {
     block_id: IdGenerator,
     /// Set of expressions that have associated temporary variables
     named_expressions: crate::NamedExpressions,
+    /// List of unused structs that have dynamic-sized arrays in them
+    ///
+    /// This is used to prune out the unsized structs that are not used by globals accessed by the
+    /// current entrypoint. This is done to make sure that, if an entrypoint doesn't need SSBOs,
+    /// that it doesn't require them and cause shader compilation failure on platforms where they
+    /// are not supported.
+    unused_unsized_types: Vec<Handle<crate::Type>>,
```

Comment:
If we are looking only for globals, maybe it would be easier to use `Vec<Handle<GlobalVariable>` ?

Reply:
The only time we use that struct now is actually [here](https://github.com/gfx-rs/naga/pull/1122/files#diff-66bc1c8d2690a79f6feff4967c60eb32e58447d0bf01fb83d9ee45de9720026bR227) where the required feature check is looping through `Handle<crate::Type>`'s, so I using handles to types is right.